bin/wasm-node/rust/src/network_service.rs

```diff
@@ -237,18 +237,25 @@ impl NetworkService {
     }
 
     /// Announces transaction to the peers we are connected to.
-    // TODO: more docs
-    pub async fn announce_transaction(
-        self: Arc<Self>,
-        transaction: &[u8],
-    ) -> Result<(), QueueNotificationError> {
+    /// Returns an error if we aren't connected to any peer, or if we fail to send the transaction to all peers.
+    pub async fn announce_transaction(self: Arc<Self>, transaction: &[u8]) -> Result<(), ()> {
```

Comment:
```suggestion
    pub async fn announce_transaction(self: Arc<Self>, transaction: &[u8]) -> Result<(), ()> {  // TODO: return a proper Error type
```

Don't remove TODOs, they're here to say what needs to change.
Or you can do the change already by creating an enum. Again, this enum should be documented.

Reply:
I opted to return the last returned error (`QueueNotificationError`) in case we have one.